utils.go

```diff
@@ -47,12 +150,86 @@ func (s *Storage) formatError(op string, err error, path ...string) error {
 	}
 }
 
+func (s *Service) formatError(op string, err error, name string) error {
+	if err == nil {
+		return nil
+	}
+
+	return services.ServiceError{
+		Op:       op,
+		Err:      err,
+		Servicer: s,
+		Name:     name,
+	}
+}
+
 // formatError converts errors returned by SDK into errors defined in go-storage and go-service-*.
 // The original error SHOULD NOT be wrapped.
 func formatError(err error) error {
 	if _, ok := err.(services.InternalError); ok {
 		return err
 	}
 
-	panic("implement me")
+	e, ok := err.(*bce.BceServiceError)
+	if ok {
+		switch e.Code {
+		case "AccessDenied":
+			return fmt.Errorf("%w, %v", services.ErrPermissionDenied, err)
+		case "NoSuchKey":
+			return fmt.Errorf("%w, %v", services.ErrObjectNotExist, err)
+		default:
+			return fmt.Errorf("%w, %v", services.ErrUnexpected, err)
+		}
+	}
+
+	return fmt.Errorf("%w, %v", services.ErrUnexpected, err)
+}
+
+func (s *Service) newStorage(pairs ...types.Pair) (store *Storage, err error) {
+	opt, err := parsePairStorageNew(pairs)
+	if err != nil {
+		return nil, err
+	}
+
+	store = &Storage{
+		client:  s.service,
+		bucket:  opt.Name,
+		workDir: "/",
+	}
+
+	if opt.HasWorkDir {
+		if !strings.HasSuffix(opt.WorkDir, "/") {
+			opt.WorkDir += "/"
+		}
```

Comment:
Maybe `WorkDir` doesn't need the suffix `/`, as it's treated as prefix for object storage.

Reply:
I think you're right.

Reply:
For object service, the absolute path is `prefix` + `path`. And the `prefix` does not necessarily contain `/`, it's generally used in `ListObject` and could be `abc`, `abc/efg`, `abc/`.

Reply:
Got it!